kore/src/Kore/Step/Step.hs

```diff
@@ -438,7 +439,61 @@ recoveryFunctionLikeResults
     => Pattern (Target variable)
     -> unifier (Results variable)
     -> unifier (Results variable)
-recoveryFunctionLikeResults _ = id
+recoveryFunctionLikeResults initial unifier = do
```

Comment:
Could you add a pointer to the document explaining this recovery process?

Also, tests?